Shared/Samples/Apply hillshade renderer to raster/ApplyHillshadeRendererToRasterView.swift

```diff
@@ -0,0 +1,93 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct ApplyHillshadeRendererToRasterView: View {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    class Model: ObservableObject {
+        /// The map that will be shown.
+        let map: Map
+        
+        /// The raster layer in the map.
+        private let rasterLayer: RasterLayer
+        
+        /// The raster renderer.
+        var renderer: HillshadeRenderer {
+            didSet {
+                // When the renderer is updated, update the layer accordingly.
+                rasterLayer.renderer = renderer
+            }
+        }
+        
+        init() {
+            // Gets the raster file URL.
+            let rasterFileURL = Bundle.main.url(forResource: "srtm", withExtension: "tiff", subdirectory: "srtm")!
+            
+            // Creates a raster with the file URL.
+            let raster = Raster(fileURL: rasterFileURL)
+            
+            // Creates a raster layer using the raster object.
+            rasterLayer = RasterLayer(raster: raster)
+            
+            // Apply the hillshade renderer to the raster layer.
+            renderer = HillshadeRenderer(
+                altitude: 45,
+                azimuth: 315,
+                slopeType: nil,
+                zFactor: 0.000016,
+                pixelSizeFactor: 1,
+                pixelSizePower: 1,
+                outputBitDepth: 8
+            )
+            rasterLayer.renderer = renderer
+            
+            // Create our map.
+            map = Map(basemap: .init(baseLayer: rasterLayer))
+        }
+    }
+    
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// A boolean value indicating if the settings panel is presented.
+    @State private var isSettingsPanelPresented: Bool = false
+    
+    var body: some View {
+        // Creates a map view to display the map.
+        MapView(map: model.map)
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Button {
+                        isSettingsPanelPresented = true
+                    } label: {
+                        Text("Settings")
+                    }
+                    .popover(isPresented: $isSettingsPanelPresented, arrowEdge: .bottom) {
+                        ApplyHillshadeRendererToRasterView.SettingsView(
+                            renderer: $model.renderer
+                        )
+                        .presentationDetents([.medium])
+                        .frame(idealWidth: 320, idealHeight: 380)
+                    }
+                }
+            }
+    }
+}
+
+#Preview {
+    ApplyHillshadeRendererToRasterView()
+}
```

Comment:
```suggestion
```
ODR doesn't work with SwiftUI previews, unfortunately. So we omit previews for samples that use offline data.

Reply:
bummer.